src/fetch_cnpj_info.py

```diff
@@ -0,0 +1,114 @@
+from concurrent import futures
+import json
+import numpy as np
+import os.path
+import pandas as pd
+import pickle
+import shutil
+from urllib.error import HTTPError
+from urllib.request import urlopen
+
+INFO_DATASET_PATH = 'data/cnpj_info.xz'
+TEMP_PATH = 'data/cnpj_info'
+
+def load_info_dataset():
+    if os.path.exists(INFO_DATASET_PATH):
+        return pd.read_csv(INFO_DATASET_PATH)
+    else:
+        return pd.DataFrame(columns=['atividade_principal',
+                                     'data_situacao',
+                                     'tipo',
+                                     'nome',
+                                     'telefone',
+                                     'atividades_secundarias',
+                                     'situacao',
+                                     'bairro',
+                                     'logradouro',
+                                     'numero',
+                                     'cep',
+                                     'municipio',
+                                     'uf',
+                                     'abertura',
+                                     'natureza_juridica',
+                                     'fantasia',
+                                     'cnpj',
+                                     'ultima_atualizacao',
+                                     'status',
+                                     'complemento',
+                                     'email',
+                                     'efr',
+                                     'motivo_situacao',
+                                     'situacao_especial',
+                                     'data_situacao_especial'])
+
+def read_csv(name):
+    return pd.read_csv('data/2016-08-08-%s.xz' % name,
+                       parse_dates=[16],
+                       dtype={'document_id': np.str,
+                              'congressperson_id': np.str,
+                              'congressperson_document': np.str,
+                              'term_id': np.str,
+                              'cnpj_cpf': np.str,
+                              'reimbursement_number': np.str})
+
+def remaining_cnpjs(info_dataset):
+    datasets = [read_csv(name)
+                for name in ['current-year', 'last-year', 'previous-years']]
+    dataset = pd.concat(datasets)
+    del(datasets)
+    is_cnpj = dataset['cnpj_cpf'].str.len() == 14.
+    cnpj_list = set(dataset.loc[is_cnpj, 'cnpj_cpf'])
+    already_fetched = set(info_dataset['cnpj'].str.replace(r'[./-]', ''))
+    return list(cnpj_list - already_fetched)
```

Comment:
I got it wrong in my first suggestion. In order to get it to work it should be:

```python
for cnpj in cnpj_list:
    if cnpj not in already_fetched:
        yield cnpj
```

Instead of:

```python
yield from (cnpj in cnpj_list if cnpj not in already_fetched)
```